### PR TITLE
[KeyVault] adding version 2018-02-14-preview with Network ACLs

### PR DESCRIPTION
**Pull request by schaabs**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by schaabs** [comment]
Creating this PR to review the new API version and Network ACL changes.  This API version is not yet deployed to production.  Until it's available this PR should not be merged.  Once it is available it will remove the DoNotMerge tag.  cc: @tiffanyachen @RandalliLama

File: specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2018-02-14/keyvault.json [hunk 1]
```diff
@@ -0,0 +1,1359 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2018-02-14-preiview",
```
Comment (mcardosos) [inline specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2018-02-14/keyvault.json]: >preiview [](start = 27, length = 8)

preview :) ?

File: specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2018-02-14/examples/createVault.json [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+                            "get",
+                            "list",
+                            "create",
+                            "update",
+                            "import",
+                            "delete",
+                            "backup",
+                            "restore",
+                            "recover",
+                            "purge"
```
Comment (dragav) [inline specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2018-02-14/examples/createVault.json]: I would not include the 'purge' permission in examples. It should be expressly and judiciously assigned, and samples have a tendency of being copied and pasted.

File: specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2018-02-14/keyvault.json (hunk 1 above)
Comment (mcardosos) [reply specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2018-02-14/keyvault.json]: All three files have this typo

---
In reply to: [168832029](https://github.com/Azure/azure-rest-api-specs/pull/2509#discussion_r168832029) [](ancestors = 168832029)

File: specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2018-02-14/examples/createSecret.json [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            }
+        },
+        "201": {
+            "body": {
+                "id": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/sample-group/providers/Microsoft.KeyVault/vaults/sample-vault/secrets/secret-name",
+                "name": "secret-name",
+                "type": "Microsoft.KeyVault/vaults/secrets",
+                "location": "westus",
+                "properties": {
+                    "attributes": {
```
Comment (dragav) [inline specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2018-02-14/examples/createSecret.json]: (nitpick) recovery level attribute..

File: specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2018-02-14/keyvault.json [hunk 4]
```diff
@@ -0,0 +1,1359 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2018-02-14-preview",
```
Comment (mcardosos) [inline specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2018-02-14/keyvault.json]: >2018-02-14-preview [](start = 16, length = 18)

Folder should have the same name as version here

File: specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2018-02-14/keyvault.json [hunk 5]
```diff
@@ -0,0 +1,1359 @@
+{
```
Comment (mcardosos) [inline specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2018-02-14/keyvault.json]: This files should be referenced on the readme.md file

File: specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2018-02-14/keyvault.json (hunk 4 above)
Comment (mcardosos) [inline specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2018-02-14/keyvault.json]: >2018-02-14-preview [](start = 16, length = 18)

Is this based on an already exisiting API version?

**Review by dragav — APPROVED** [review]

**Review by msfcolombo — APPROVED** [review]

File: specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2018-02-14/keyvault.json (hunk 4 above)
Comment (schaabs) [reply specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2018-02-14/keyvault.json]: Is this required?  I thought having preview in both the path and the folder name would be redundant.  I followed the pattern I used in the data-plane:

https://github.com/Azure/azure-rest-api-specs/tree/master/specification/keyvault/data-plane/Microsoft.KeyVault/preview

File: specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2018-02-14/keyvault.json (hunk 5 above)
Comment (schaabs) [reply specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2018-02-14/keyvault.json]: Added.

File: specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2018-02-14/keyvault.json (hunk 4 above)
Comment (schaabs) [reply specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2018-02-14/keyvault.json]: yes this is based of 2016-10-01 with the addition of the networkAcls property on VaultProperties

Comment (mcardosos) [reply specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2018-02-14/keyvault.json]: Yes, it is required

Comment (mcardosos) [reply specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2018-02-14/keyvault.json]: Then, could you please submit the PR like this...
- First commit, copy and paste swaggers from already existing version.
- Second commit, modify the pasted swaggers so they now reflect the new API.
This way it is a lot easier to review, I can just look at the diffs :)

**Comment by schaabs** [comment]
Closing this PR as we are going to merge these changes in the keyvault_preview branch rather than master to better facilitate PR automation

**Pull request closed without merging**